packages/inspector-extension/src/index.ts

```diff
@@ -56,55 +53,52 @@ const inspector: JupyterLabPlugin<IInspector> = {
       namespace
     });
 
-    /**
-     * Create and track a new inspector.
-     */
-    function newInspectorPanel(): InspectorPanel {
-      const inspector = new InspectorPanel();
-
-      inspector.id = 'jp-inspector';
-      inspector.title.label = 'Inspector';
-      inspector.disposed.connect(() => {
-        if (manager.inspector === inspector) {
-          manager.inspector = null;
-        }
-      });
-
-      // Track the inspector.
-      let widget = new MainAreaWidget({ content: inspector });
-      tracker.add(widget);
-
-      // Add the default inspector child items.
-      Private.defaultInspectorItems.forEach(item => {
-        inspector.add(item);
-      });
-
-      return inspector;
+    let source: IInspector.IInspectable | null = null;
+    let inspector: MainAreaWidget<InspectorPanel>;
+    function createInspector(): void {
+      if (!inspector || inspector.isDisposed) {
+        inspector = new MainAreaWidget({ content: new InspectorPanel() });
+        inspector.id = 'jp-inspector';
+        inspector.title.label = 'Inspector';
+        tracker.add(inspector);
+        source = source && !source.isDisposed ? source : null;
+        inspector.content.source = source;
+      }
+      if (!inspector.isAttached) {
+        shell.addToMainArea(inspector, { activate: false });
+      }
+      shell.activateById(inspector.id);
     }
 
+    // Add command to registry and palette.
+    commands.addCommand(command, {
+      label,
+      execute: () => {
+        createInspector();
+      }
+    });
+    palette.addItem({ command, category });
```

Comment:
I wonder if also adding this to the launcher and file menus would make it more discoverable. I know that lots of people are unaware that the inspector exists.

Reply:
Good point. I added it to  the "File" menu and to the launcher. (**screenshot** in main PR description)